test/Microsoft.AspNetCore.Razor.Test.Common/Language/TestTagHelperDescriptorBuilderExtensions.cs

```diff
@@ -14,7 +14,109 @@ public static TagHelperDescriptorBuilder TypeName(this TagHelperDescriptorBuilde
                 throw new ArgumentNullException(nameof(builder));
             }
 
-            return builder.SetTypeName(typeName);
+            builder.SetTypeName(typeName);
+
+            return builder;
+        }
+
+        public static TagHelperDescriptorBuilder DisplayName(this TagHelperDescriptorBuilder builder, string displayName)
+        {
+            if (builder == null)
+            {
+                throw new ArgumentNullException(nameof(builder));
+            }
+
+            builder.DisplayName = displayName;
+
+            return builder;
+        }
+
+        public static TagHelperDescriptorBuilder AllowChildTag(this TagHelperDescriptorBuilder builder, string allowedChild)
+        {
+            if (builder == null)
+            {
+                throw new ArgumentNullException(nameof(builder));
+            }
+
+            builder.AllowedChildTags.Add(allowedChild);
+
+            return builder;
+        }
+
+        public static TagHelperDescriptorBuilder TagOutputHint(this TagHelperDescriptorBuilder builder, string hint)
+        {
+            if (builder == null)
+            {
+                throw new ArgumentNullException(nameof(builder));
+            }
+
+            builder.TagOutputHint = hint;
+
+            return builder;
+        }
+
+        public static TagHelperDescriptorBuilder Documentation(this TagHelperDescriptorBuilder builder, string documentation)
+        {
+            if (builder == null)
+            {
+                throw new ArgumentNullException(nameof(builder));
+            }
+
+            builder.Documentation = documentation;
+
+            return builder;
+        }
+
+        public static TagHelperDescriptorBuilder AddMetadata(this TagHelperDescriptorBuilder builder, string key, string value)
+        {
+            if (builder == null)
+            {
+                throw new ArgumentNullException(nameof(builder));
+            }
+
+            builder.Metadata[key] = value;
+
+            return builder;
+        }
+
+        public static TagHelperDescriptorBuilder AddDiagnostic(this TagHelperDescriptorBuilder builder, RazorDiagnostic diagnostic)
+        {
+            if (builder == null)
+            {
+                throw new ArgumentNullException(nameof(builder));
+            }
+
+            builder.Diagnostics.Add(diagnostic);
+
+            return builder;
+        }
+
+        public static TagHelperDescriptorBuilder BoundAttributeDescriptor(
```

Comment:
Note: I had to rename this and similar methods to something other that what we have in the product code to avoid collision. The compiler cannot handle overloads that are of the same name, arguments but different return type.